src/lib/dialog/dialog-content-directives.ts

```diff
@@ -17,6 +17,9 @@ export class MdDialogClose {
   /** Screenreader label for the button. */
   @Input('aria-label') ariaLabel: string = 'Close dialog';
 
+  /** Result to return to the dialog opener. */
+  @Input('md-dialog-close') result: any;
```

Comment:
This doesn't work in compatibility mode. We need something for `mat-dialog-close` too.